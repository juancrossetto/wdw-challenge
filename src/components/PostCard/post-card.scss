@use '../../scss/' as *;

.post-card {
  background-color: $white;
  height: 350px;
  width: 350px;
  max-width: 350px;
  position: relative;
  margin: 0 auto;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  @include for-phone-only {
    width: 100%;
    max-width: 335px;
    min-width: 250px;
    height: 100%;
    max-height: 342px;
    min-height: 250px;
  }
  &__img {
    height: 100%;
    max-height: 280px;
    @include for-phone-only {
      max-height: 273px;
    }
    img {
      width: 100%;
      height: 100%;
      border-top-left-radius: 10px;
      border-top-right-radius: 10px;
    }
  }

  &__footer {
    background-color: $gray-two;
    height: 70px;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
    @include flex(center, flex-start);

    &__avatar {
      width: 45px;
      height: 45px;
      border-radius: 50%;
      margin-right: 11px;
      margin-left: 12px;
    }
    &__texts {
      @include flex(flex-start, space-between);
      flex-direction: column;

      &__title {
        color: $dark;
        font-style: normal;
        font-weight: 500;
        font-size: 16px;
        line-height: 19px;
        letter-spacing: 0.01em;
        margin-bottom: 4px;
      }

      &__hashtag {
        color: $black;
        font-style: normal;
        font-weight: 400;
        font-size: 16px;
        line-height: 19px;

        letter-spacing: 0.01em;
      }
    }
  }
}

.container {
  max-width: $max-width-desktop;
  margin: auto;
}
