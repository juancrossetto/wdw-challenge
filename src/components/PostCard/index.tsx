import React, { FC } from 'react';
import './post-card.scss';

interface PostCardProps {
  post: IPost;
}
const PostCard: FC<PostCardProps> = ({ post }) => {
  return (
    <div className="post-card">
      <div className="post-card__img">
        <img src={post.img} alt={`Post image of ${post.title}`} />
      </div>
      <div className="post-card__footer">
        <img className="post-card__footer__avatar" src={post.avatarImg} alt={`Avatar image of ${post.title}`} />
        <div className="post-card__footer__texts">
          <p className="post-card__footer__texts__title">{post.title}</p>
          <span className="post-card__footer__texts__hashtag">{post.hashTag}</span>
        </div>
      </div>
    </div>
  );
};

export default PostCard;
