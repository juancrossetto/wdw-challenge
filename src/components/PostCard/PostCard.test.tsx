import React from 'react';
import renderer from 'react-test-renderer';

import PostCard from '.';

describe('Post Card component', () => {
  const postMocked: IPost = { hashTag: '#test', title: 'title test', avatarImg: 'avatar.png', img: 'bgTest.png' };
  test('Post Card render correctly', () => {
    const tree = renderer.create(<PostCard post={postMocked} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
