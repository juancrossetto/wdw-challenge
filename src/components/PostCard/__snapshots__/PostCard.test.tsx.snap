// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Card component Post Card render correctly 1`] = `
<div
  className="post-card"
>
  <div
    className="post-card__img"
  >
    <img
      alt="Post image of title test"
      src="bgTest.png"
    />
  </div>
  <div
    className="post-card__footer"
  >
    <img
      alt="Avatar image of title test"
      className="post-card__footer__avatar"
      src="avatar.png"
    />
    <div
      className="post-card__footer__texts"
    >
      <p
        className="post-card__footer__texts__title"
      >
        title test
      </p>
      <span
        className="post-card__footer__texts__hashtag"
      >
        #test
      </span>
    </div>
  </div>
</div>
`;
